File: model/src/test/java/pl/agh/harmonytools/model/KeyTest.java
package pl.agh.harmonytools.model;


import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class KeyTest {

    @Test
    public void baseNoteTest() {
        Key G = new Key("G##");
        Assertions.assertEquals(G.getBaseNote(), BaseNote.G);
    }

    @Test
    public void notSupportedKeyTest() {
        Assertions.assertThrows(IllegalArgumentException.class, () -> new Key("R#"));
    }
}

File: model/src/test/scala/pl/agh/harmonytools/model/KeyTest.scala
package pl.agh.harmonytools.model

import org.scalatest.{FunSuite, Matchers}
import pl.agh.harmonytools.model.key.Key
import pl.agh.harmonytools.model.note.BaseNote

class KeyTest extends FunSuite with Matchers {
  test("base note test") {
    val G = Key("g##")
    G.baseNote shouldBe BaseNote.G
  }

  test("not supported key test") {
    assertThrows[IllegalArgumentException](Key("R#"))
  }
}
